File: src/main/java/io/myosotisdev/minestom/permission/PermissionGroup.java
package io.myosotisdev.minestom.permission;

import java.util.Collection;
import java.util.HashSet;
import java.util.Set;

public class PermissionGroup implements IPermissionHandler
{
    private String          groupId;
    private Set<Permission> permissions;

    public PermissionGroup(String groupId)
    {
        this.groupId = groupId;
        this.permissions = new HashSet<>();
    }

    public String getId()
    {
        return this.groupId;
    }

    @Override
    public void addPermission(Permission perm)
    {
        this.permissions.add(perm);
    }

    @Override
    public void removePermission(Permission perm)
    {
        this.permissions.remove(perm);
    }

    @Override
    public boolean hasPermission(Permission perm)
    {
        return this.permissions.stream().anyMatch(permission -> permission.matchWith(perm));
    }

    @Override
    public Collection<Permission> permissions()
    {
        return this.permissions;
    }
}

File: src/main/java/io/myosotisdev/minestom/permission/PermissionGroup.kt
package io.myosotisdev.minestom.permission

import com.google.gson.JsonObject
import com.google.gson.JsonParser
import java.io.File
import java.io.FileReader
import java.util.*
import kotlin.collections.HashSet

class PermissionGroup : IPermissionHandler
{
    private val permissions: MutableSet<Permission?> = HashSet()
    val id: String;
    var name: String = ""

    constructor(id: String)
    {
        this.id = id
        this.name = id;
    }

    constructor(id: String, file: File) : this(id)
    {
        var jsonObject: JsonObject = JsonParser.parseReader(FileReader(file)).asJsonObject

    }

    override fun addPerm(perm: Permission?)
    {
        permissions.add(perm)
    }

    override fun removePerm(perm: Permission?)
    {
        permissions.remove(perm)
    }

    override fun hasPerm(perm: Permission?): Boolean
    {
        return permissions.stream()
                .anyMatch { permission: Permission? -> permission!!.matchWith(perm!!) }
    }

    override fun permissions(): Collection<Permission?>?
    {
        return permissions
    }
}